app/uk/gov/hmrc/apisubscriptionfields/model/Decoder.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apisubscriptionfields.model
+
+trait Decoder[T] {
+  val separator: SeparatorType
+  val numOfParts: Int
+  protected def decode(tokens: Seq[String]): T
+
+  def decode(text:String): Option[T] = {
+    def findSeparator(separatorToFind: SeparatorType) : SeparatorType = {
+      if (text.split(separatorToFind).length > numOfParts)
+        findSeparator(separatorToFind+separator)
```

Comment:
add space between operator and variables:
```
findSeparator(separatorToFind + separator)
```

Reply:
Done.